tests/test_rects.py

```diff
@@ -0,0 +1,231 @@
+
+
+def test_rects_extraction_code_inline(tmpdir: PathLike) -> None:
+    """Test rect extraction of inline codeblock."""
+    smart_page_crop = (
+        True  # remove header and footers so rects IN chapters are left only.
+    )
+
+    objects = libpdf.load(
+        PDF_RECTS_EXTRACTION,
+        smart_page_crop=smart_page_crop,
+        visual_debug=False,
+        visual_debug_output_dir=tmpdir.join("visual_debug_dir"),
+        visual_split_elements=True,
+    )
+    assert objects.flattened.rects is not None
+
+    chapter = find_chapter(objects, "Code Inline Highlighting")
+
+    # 2 inline code blocks, but the first one is broken in two lines
+    assert check_chapter_rects_count(chapter) == 1 * 3
+
+    paragraph = check_chapter_contains_text_paragraph(
+        chapter, "from pathlib import Path"
+    )
+    assert paragraph is not None
+
+    rect = check_chapter_contains_text_rect(chapter, "from pathlib import Path")
+    assert rect is not None
+    assert rect.textbox.text == "from pathlib import Path"
+    assert check_content_color(rect, (0.945098, 0.945098, 0.945098))
+    assert check_content_margins_greater(rect, paragraph, 234)
+
+    assert (
+        check_chapter_contains_text_rect(
+            chapter, "decode_title(obj_bytes: bytes) -> str"
+        )
+        is None
+    )
+    rect = check_chapter_contains_text_rect(chapter, "decode_title(obj_bytes: bytes)")
+    assert rect is not None
+    rect_str = check_chapter_contains_text_rect(chapter, "str ")
+    assert rect_str is not None
+
+    assert rect_str.textbox.x0 < rect.textbox.x0
+
+
+def test_rects_extraction_adminition() -> None:
+    """Test rect extraction of 3 admonitions."""
+    smart_page_crop = (
+        True  # remove header and footers so rects IN chapters are left only.
+    )
+
+    objects = libpdf.load(PDF_RECTS_EXTRACTION, smart_page_crop=smart_page_crop)
+    assert objects.flattened.rects is not None
+
+    chapter = find_chapter(objects, "Adminition")
+
+    assert (
+        check_chapter_rects_count(chapter) == 3 * 2
+    )  # 2 rects per admonition, 3 types of admonition
+
+    rect = check_chapter_contains_text_rect(chapter, "A very importing Adminition")
+    assert rect is not None
+    assert check_content_color(rect, (0.858824, 0.980392, 0.956863))
+
+    rect_inner = check_chapter_contains_text_rect(chapter, "Wichtig")
+    assert rect_inner is not None
+    assert check_content_color(rect, (0.858824, 0.980392, 0.956863))
+
+
+def test_rects_extraction_table(tmpdir: PathLike) -> None:
+    """Test rect extraction of table colored cells."""
+    smart_page_crop = (
+        True  # remove header and footers so rects IN chapters are left only.
+    )
+
+    objects = libpdf.load(
+        PDF_RECTS_EXTRACTION,
+        smart_page_crop=smart_page_crop,
+        visual_debug=False,
+        visual_debug_output_dir=tmpdir.join("visual_debug_dir"),
+        visual_split_elements=True,
```

Comment:
these 2 should not be needed if you set `visual_debug=False`

Reply:
Still don't understand this. Why would you provide fields that have no meaning in this context?
![image](https://github.com/useblocks/libpdf/assets/839934/8b3f7fad-6caa-49cd-8463-7bcd613fa01e)
The libpdf run will not populate this directory nor split elements, so why passing the params. Maybe I misunderstand your use case here. :thinking: